docs/source/daml/reference/contract-keys.rst

```diff
@@ -19,76 +19,76 @@ Here's an example of setting up a contract key for a bank account, to act as a b
 
 .. _fetchbykey:
 
-``fetchByKey``
+fetchByKey
 ==============
 
 ``(fetchedContractId, fetchedContract) <- fetchByKey @ContractType contractKey``
 
-Use ``fetchByKey`` to fetch the ID and data of the contract with the specified key. It is an alternative to the currently-used ``fetch``.
+Use ``fetchByKey`` to fetch the ID and data of the contract with the specified key. It is an alternative to ``fetch`` and behaves the same in most ways. 
 
 It returns a tuple of the ID and the contract object (containing all its data).
 
-``fetchByKey`` is authorized like ``fetch`` so it needs to be
-authorized by at least one stakeholder. There are no restrictions on
-the submitter.
+Like ``fetch``, ``fetchByKey`` needs to be authorized by at least one stakeholder.
 
 ``fetchByKey`` fails and aborts the transaction if:
 
-- Missing authorization, i.e., no authorization from a
-  stakeholder of the contract you are trying to fetch.
+- The submitting party is not a stakeholder on a contract with the givem key, or
+- A contract was found, but the ``fetchByKey`` violates the authorization rule, meaning no stakeholder authorized the ``fetch``..
 
-This means that if it fails, it doesn't guarantee that a contract with that key doesn't exist, just that you can't see one.
+This means that if it fails, it doesn't guarantee that a contract with that key doesn't exist, just that the submitting party doesn't know about it, or there are issues with authorization.
 
 Because different templates can use the same key type, you need to specify the type of the contract you are trying to fetch using the ``@ContractType`` syntax.
 
 .. _lookupbykey:
 
-``lookupByKey``
+lookupByKey
 ===============
 
 ``optionalContractId <- lookupByKey @ContractType contractKey``
 
 Use ``lookupByKey`` to check whether a contract with the specified key exists. If it does exist, ``lookupByKey`` returns the ``Some contractId``, where ``contractId`` is the ID of the contract; otherwise, it returns ``None``.
 
-``lookupByKey`` needs to be authorized by **all** maintainers of the
-contract you are trying to lookup. There are no restrictions on the
-submitter.
+``lookupByKey`` needs to be authorized by **all** maintainers of the contract you are trying to lookup. The reason for that is denial of service protection. WIthout this restriction, a malicious Party could legitimately request another party to validate any number of negative contract key lookups without that other party ever having signed anything.
 
-If the lookup fails (ie returns ``None``), this guarantees that no contract has this key.
+Unlike ``fetchByKey``, the transaction **does not fail** if no contract with the given key is found. Instead, ``lookupByKey`` just returns ``None``. However, that does not guarantee that no such contract exists. The submitting party may simply not know about it.
```

Comment:
What happens if such a key does exist? Then validation fails? We should mention that here.